### PR TITLE
Web SDK UI customisation: buttons

### PR DESCRIPTION
**Pull request by kopijunkie**
# Feature
We want to offer customers some supported customisation options for the Web SDK so that they can customise the UI of the Primary and Secondary Action Buttons as well as the Document Type Button. 

The SDK's Primary and Secondary Buttons have also been changed to use the Button component from Onfido's Design System, [Castor](https://github.com/onfido/castor) from PR #1289 

In this PR, the following button attributes are customisable 

for **Primary and Secondary Action Buttons**:

- Background & border colour
- Border radius 
- Text/label colour
- Focus/Active state (background colour only)
- Hover state (background colour only)

Screenshot 2021-02-17 at 16 38 53

Additionally for button groups there is the option to not display button groups inline which is the current default.

Screenshot 2021-02-17 at 16 39 26

for **Document Type Buttons**:

- Border radius
- Border color
- Focus/Active state (border colour only)
- Hover state (border colour only)

Screenshot 2021-02-17 at 16 30 13 

When a user in on the Cross Device flow and is continuing the IDV process on their mobile devices, these UI customisations will carry over to the Web SDK there also. This is currently not possible if customisation is done by overriding the existing Web SDK styles in the host app/site's stylesheets.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [x] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "check:es": "eslint '{src,test}/**/*.{js,jsx,ts,tsx}'",
     "check:styles": "stylelint 'src/**/*.scss'",
     "check:format": "prettier . -c",
+    "format": "npx prettier --write .",
```
Comment (josokinas) [inline package.json:27]: No need for `npx` here, it's already referred to local `.bin`:

```suggestion
    "format": "prettier --write .",
```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/App/ModalApp.tsx [hunk 2]
```diff
@@ -174,6 +177,10 @@ class ModalApp extends Component<Props> {
       const validEnterpriseFeatures = getEnterpriseFeaturesFromJWT(token)
       this.setConfiguredEnterpriseFeatures(validEnterpriseFeatures, options)
     }
+
+    if (customUI && customUI !== prevCustomUI) {
```
Comment (josokinas) [inline src/components/App/ModalApp.tsx:181]: I think you would want to deep compare objects here, so probably something like using https://lodash.com/docs/4.17.15#isEqual or equivalent.

File: src/components/Theme/tokens.scss [hunk 3]
```diff
@@ -21,46 +40,77 @@
...
+  );
+  --color-border-doc-type-button-active: rgba(
+    var(--ods-color-border-action-focus)
+  );
+
+  /* Back, Close (modal) icon button colors */
+  --color-background-button-icon-hover: rgba(var(--ods-color-neutral-400));
+  --color-background-button-icon-active: rgba(var(--ods-color-neutral-600));
+
+  /* Selfie Camera button colors */
```
Comment (josokinas) [inline src/components/Theme/tokens.scss:110]: These are not documented elsewhere, is it expected?

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (rabelloo) [reply src/components/App/ModalApp.tsx:181]: First, please no Lodash.
SDK is size sensitive and Lodash weighs a ton.

Second, and most importantly, I'd argue deep compare is a bad idea.
Not only because it is computationally expensive, but mainly because it encourages object mutations by supporting it.
Immutability and reactivity, I say.

File: src/components/Theme/style.scss [hunk 4]
```diff
@@ -6,40 +6,42 @@
...
 
       &:disabled {
-        background-color: var(--color-primary-button-disabled);
+        background-color: var(--color-background-button-primary-disabled);
+        border-color: $color-transparent;
       }
 
-      &:hover {
-        background-color: var(--color-primary-button-hover);
+      &:hover:not(:disabled) {
```
Comment (rabelloo) [inline src/components/Theme/style.scss:26]: No need for `:not()` here, simply move it before `&:disabled` and it will be overriden.

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (josokinas) [reply src/components/App/ModalApp.tsx:181]: Lodash was just an example.

But isn't `customUI !== prevCustomUI` always truthy though? React will always emit a new object, isn't it?

File: src/components/Theme/tokens.scss (hunk 3 above)
Comment (rabelloo) [reply src/components/Theme/tokens.scss:110]: They don't have `--osdk` prefix so I think they're internal

File: src/components/Theme/utils.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { kebabCase } from '~utils/string'
+import { UICustomisationOptions } from '~types/ui-customisation-options'
+
+export const setUICustomisations = (customUI: UICustomisationOptions): void => {
+  const sdkCustomisations = Object.keys(customUI).map(
+    (property: keyof typeof customUI) =>
+      `--osdk-${kebabCase(property)}: ${customUI[property]};`
+  )
```
Comment (rabelloo) [inline src/components/Theme/utils.ts:8]: 💅🏻  [nitpick]

```suggestion
  const sdkCustomisations = Object.entries(customUI).map(
    ([property, value]) => `--osdk-${kebabCase(property)}: ${value};`
  )
```

File: src/components/Theme/utils.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    document.head.insertAdjacentHTML('beforeend', style)
+  }
+}
+
+export const isButtonGroupStacked = (): boolean =>
+  !!JSON.parse(
+    getComputedStyle(document.body).getPropertyValue(
+      '--osdk-button-group-stacked'
+    )
+  )
```
Comment (rabelloo) [inline src/components/Theme/utils.ts:25]: 💅🏻  [nitpick]

That CSS variable should be a string right? Either `false` or `true`.
Instead of the expensive `JSON.parse` you can just check the value.

```suggestion
export const isButtonGroupStacked = (): boolean =>
  'true' ===
  getComputedStyle(document.body)
    .getPropertyValue('--osdk-button-group-stacked')
    .trim()
```

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (rabelloo) [reply src/components/App/ModalApp.tsx:181]: As long as `prepareInitialStore` is always called with a pointer to the same object in memory, and `prevOptions` doesn't mutate its `customUI` object reference, they should be stable.

Which means, as long as `prepareInitialStore` isn't called with a new `customUI` in its `options`, the objects will always be the same.

**Comment by rabelloo** [comment]
Please help me understand, is there a need for an SDK customer to customise a primary button without changing other primary components?

For instance, let's say the default Castor primary color is simply `blue`.
An SDK customer requires their primary button to be `red`.
However they still want checkboxes, which use primary colors too, to remain `blue`.

Is that a use case?

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (josokinas) [reply src/components/App/ModalApp.tsx:181]: Fine, ignore me, I'm like on Redux now thinking it emits a new immutable object each time.

Comment (rabelloo) [reply src/components/App/ModalApp.tsx:181]: Even in Redux if the reducer doesn't touch `customUI` it's still the same.
The base store object might change, but the value of one of its properties can still hold a pointer to the same object.

Comment (rabelloo) [reply src/components/App/ModalApp.tsx:181]: For example:

```ts
const customUI = { foo: 'bar' };
const a = { customUI };
let b = { customUI };

a.customUI === b.customUI;
// true

b = { ...b, customUI };

a.customUI === b.customUI;
// true
```

File: src/types/ui-customisation-options.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+export type UICustomisationOptions = {
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/types/ui-customisation-options.ts:1]: Yeah, I always need to consciously use American English otherwise instances like this happens. xD
Will make the change.

File: src/demo/demoUtils.ts [hunk 8]
```diff
@@ -6,6 +6,9 @@ import {
   StepTypes,
 } from '~types/steps'
 import { ServerRegions, SdkOptions } from '~types/sdk'
+import { UICustomisationOptions } from '~types/ui-customisation-options'
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const customUIConfig = require('./custom-ui-config.json')
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/demo/demoUtils.ts:11]: @ethan605 are we OK to go back to using the above import syntax for JSON files now? I recall in your PRs with the partial TS conversions you mentioned to me we had to (temporarily?) use require instead.

File: package.json (hunk 1 above)
Comment (kopijunkie) [reply package.json:27]: Oh, cool. Nice to know.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/utils.ts (hunk 5 above)
Comment (kopijunkie) [reply src/components/Theme/utils.ts:8]: Nice. I'm still a TS newbie so just went with this based off what I found when looking into fixing a type error.

File: src/components/Theme/utils.ts (hunk 6 above)
Comment (kopijunkie) [reply src/components/Theme/utils.ts:24]: Yes, the CSS variable should be a string. Will double check this locally just to be sure.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by kopijunkie** [comment]
> Please help me understand, is there a need for an SDK customer to customise a primary button without changing other primary components?
> 
> For instance, let's say the default Castor primary color is simply `blue`.
> An SDK customer requires their primary button to be `red`.
> However they still want checkboxes, which use primary colors too, to remain `blue`.
> 
> Is that a use case?

That is a potential use case. But what we've actually done is break down the customisation options work into the individual UI components. This is the 1st piece of work which includes the foundation for the customer's customisations carrying over to the cross device flow. Will follow it up with PRs for links, modal background & border, text fonts & colour, icon background.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by josokinas** [comment]
> That is a potential use case.

If a requirement is to have theming per component, then it's OK.

However, if there is no such requirement - I would be pro Castor theming system, where a theme is applied to a context but not each component. That is, if changing "primary" color that change is applied to all components that use "primary" variation.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rabelloo** [comment]
> Will follow it up with PRs for links, modal background & border, text fonts & colour, icon background

My main concern is giving too much power to customers may make the design inconsistent, not to mention maintaining it very hard and time-consuming for you.

All the components you mentioned should be able to be controlled with a few simpler Castor CSS variables instead of requiring such fine-grained control.
For example:

```scss
// modal background
--ods-color-background-main: grey;
// modal border
--ods-border-radius-medium: 8px;
// text font
font-family: Verdana;
// text color
--ods-color-content-main: black;
```

Controlling the theme at a higher level rather the component at a lower one is simpler precisely because it offers more consistency over flexibility.
I don't think anyone reasonably wants primary buttons to have a different color than primary checkboxes or radios, for example.

Also, it'll make theming much easier for you as you don't need to map every single component and their properties, you only need to ensure they use proper CSS variables (tokens).

**TL;DR:** I'd recommend mapping the `config.customUI` values directly to Castor CSS variables and giving control over theme, not components.
If necessary due to lack of Castor support (like Icon background color), then create SDK specific CSS variable that extends Castor.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/utils.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    ([property, value]) => `--osdk-${kebabCase(property)}: ${value};`
+  )
+  if (sdkCustomisations.length > 0) {
+    const style = `
+      <style>
+          :root {
+            ${sdkCustomisations.join('\n')}
+          }
+      </style>`
+    document.head.insertAdjacentHTML('beforeend', style)
```
Comment (stefaniacardenas) [inline src/components/Theme/utils.ts:15]: Is it safe to use `insertAdjacentHTML` like this?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/App/ModalApp.tsx [hunk 10]
```diff
@@ -25,6 +25,8 @@ import type {
   StepConfigDocument,
   DocumentTypes,
 } from '~types/steps'
+import { kebabCase } from '~utils/string'
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/components/App/ModalApp.tsx]: I missed this in my IDE. But yeah, I've noticed that our linter checks, IDE auto formatting don't seem to work on the `.ts(x)` files with the TypeScript conversions @ethan605 did. :\

File: src/components/Theme/utils.ts (hunk 9 above)
Comment (kopijunkie) [reply src/components/Theme/utils.ts:15]: I think so for this use case and the string is escaped. Additionally none of the CSS/Sass these CSS variables are applied to are for `content`.

File: README.md [hunk 11]
```diff
@@ -388,9 +391,42 @@ A number of options are available to allow you to customise the SDK:
...
+  | --------------------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+  | `borderRadiusButton`                    | `string`   | Change border radius value of Primary, Secondary and Document Type Option buttons                                                           |
+  | `buttonGroupStacked` (default: `false`) | `boolean`  | Display Primary, Secondary button group in Document and Face capture confirmation screens are in separate rows instead of inline by default |
+
 - **`steps {List} optional`**
 
-  List of the different steps and their custom options. Each step can either be specified as a string (when no customisation is required) or an object (when customisation is required):
+  List of the different steps and their custom options. Each step can either be specified as a string (when no customization is required) or an object (when customization is required):
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply README.md:429]: Looks like we should be keeping in mind to use American English for all documentation.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/demo/demoUtils.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (ethan605) [reply src/demo/demoUtils.ts:11]: @kopijunkie I got some problems with importing `json` files in the past, but I just tested again and it seems that it works out of the box now. You can safely convert this into ES6 import now.

**Comment by kopijunkie** [comment]
@josokinas @rabelloo As discussed on Slack, we want to give a greater level of control over the Web SDK UI customisation to integrators with default mappings to Castor theme/base tokens.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/demo/demoUtils.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/demo/demoUtils.ts:11]: Nice. Will change to use the ES6 import instead then.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/types/sdk.ts [hunk 12]
```diff
@@ -1,6 +1,7 @@
 import type { SupportedLanguages, LocaleConfig } from './locales'
 import type { StepConfig, StepTypes } from './steps'
 import type { EnterpriseFeatures } from './enterprise'
+import { UICustomizationOptions } from './ui-customisation-options'
```
Comment (josokinas) [inline src/types/sdk.ts:4]:
```suggestion
import type { UICustomizationOptions } from './ui-customisation-options'
```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/tokens.scss [hunk 13]
```diff
@@ -21,46 +40,74 @@
   --font-weight-subtitle: 600;
   --font-weight-body: 500;
 
-  /* Color Variables */
+  /* General UI element background color settings */
+  --osdk-color-border-surface-modal: rgba(var(--ods-color-border-input));
   --color-background: rgba(var(--ods-color-neutral-050));
   --color-warning: rgba(var(--ods-color-primary-500));
-  --color-error: rgba(220, 42, 42);
+  --color-error: rgba(var(--ods-color-background-negative));
```
Comment (josokinas) [inline src/components/Theme/tokens.scss:47]: No action needed, just curious why don't you use Castor Sass helper? So it's:

```suggestion
  --color-error: castor.color('background-negative');
```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Review by josokinas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/utils.ts (hunk 9 above)
Comment (stefaniacardenas) [reply src/components/Theme/utils.ts:15]: Looks fine to me but might be worth double checking with security, I have a thread with them currently regarding `innerHtml` usage.

**Review by stefaniacardenas — APPROVED** [review]

**Review by rabelloo — APPROVED** [review]

File: src/components/Theme/utils.ts (hunk 9 above)
Comment (rabelloo) [reply src/components/Theme/utils.ts:15]: If you're really worried you could refactor to manually create the node.

```ts
    const style = document.createElement('style');
    style.textContent = `:root { ... }`;
    document.head.appendChild(style);
```

You could also set it directly to the root element as CSS, which is probably as simple as you can get.

```ts
    document.documentElement.style.cssText = sdkCustomisations.join('\n');
```

File: src/components/Theme/tokens.scss (hunk 13 above)
Comment (kopijunkie) [reply src/components/Theme/tokens.scss:47]: No specific reason actually. I didn't know of the helper until it came up in the recent discussion. & then forgot to make the changes. 😅

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Review by ethan605 — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1339-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1339-pr-onfido-sdk-ui-onfido.surge.sh
